src/MediatR/Internal/StreamRequestHandlerWrapper.cs

```diff
@@ -0,0 +1,41 @@
+namespace MediatR.Internal
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.CompilerServices;
+    using System.Threading;
+
+    internal abstract class StreamRequestHandlerBase : HandlerBase
+    {
+        public abstract IAsyncEnumerable<object?> Handle(object request, CancellationToken cancellationToken, ServiceFactory serviceFactory);
+    }
+
+    internal abstract class StreamRequestHandlerWrapper<TResponse> : StreamRequestHandlerBase
+    {
+        public abstract IAsyncEnumerable<TResponse> Handle(IRequest<TResponse> request, CancellationToken cancellationToken,
+            ServiceFactory serviceFactory);
+    }
+
+    internal class StreamRequestHandlerWrapperImpl<TRequest, TResponse> : StreamRequestHandlerWrapper<TResponse>
+        where TRequest : IRequest<TResponse>
+    {
+        public async override IAsyncEnumerable<object?> Handle(object request, [EnumeratorCancellation] CancellationToken cancellationToken, ServiceFactory serviceFactory)
+        {
+            await foreach (var item in Handle((IRequest<TResponse>) request, cancellationToken, serviceFactory))
+            {
+                yield return item;
+            }
+        }
+
+        public override IAsyncEnumerable<TResponse> Handle(IRequest<TResponse> request, CancellationToken cancellationToken, ServiceFactory serviceFactory)
+        {
+            IAsyncEnumerable<TResponse> Handler() => GetHandler<IStreamRequestHandler<TRequest, TResponse>>(serviceFactory).Handle((TRequest) request, cancellationToken);
+
+            return serviceFactory
+                .GetInstances<IStreamPipelineBehavior<TRequest, TResponse>>()
+                .Reverse()
+                .Aggregate((StreamHandlerDelegate<TResponse>) Handler, (next, pipeline) => () => pipeline.Handle((TRequest) request, cancellationToken, next))();

```

Comment:
Can the `.WithCancellation(cancellationToken).ConfigureAwait(false)` (as in `GenericStreamPipelineBehavior.cs` in the stream example) be done here instead of expecting the implementation to handle it? Specifically, replace
```csharp
pipeline.Handle((TRequest) request, cancellationToken, next)
```
with
```csharp
pipeline.Handle((TRequest) request, cancellationToken, () => next().WithCancellation(cancellationToken).ConfigureAwait(false))
```

Reply:
Thanks for the review comments. Made a change in the code. Please review again because of compile errors I had to alter your suggested change a little bit.

Reply:
I did a little research and found [this instructive blog post](https://oleh-zheleznyak.blogspot.com/2020/07/enumeratorcancellation.html). I see two options:
1. Put the `EnumeratorCancellation` attribute on the cancellation token argument in the implementations of the stream handler (examples and `StreamRequestExceptionActionProcessorBehavior` and such, plus you're already doing it in the tests) and hope everyone using the library follows that practice (which they should).
2. Wrap `next()` as `() => NextWrapper(next(), cancellationToken)` with the following implementation:
```csharp
private static IAsyncEnumerable<T> NextWrapper<T>(
    IAsyncEnumerable<T> items,
    [EnumeratorCancellation] CancellationToken cancellation)
{
    var cancellable = items
        .WithCancellation(cancellationToken)
        .ConfigureAwait(false);
    await foreach (var i in cancellable)
    {
        yield return i;
    }
}
```

In either case, you should remove the `.WithCancellation(cancellationToken).ConfigureAwait(false)` from everywhere else.